docs/design/DIDManagement.md

```diff
@@ -0,0 +1,187 @@
+        - `endpoint`: a string value representing the endpoint attrib of this DID on an indy network
+- **Tags:**
+    - `method`: the method name, e.g. `peer`
+    - `verkey`: the base58 encoding of the public key associated with this DID
+    - `verkey_type`: the key type of the verkey, e.g. `ed25519`
+
+### Current State of Key Storage
+
+Keys are managed by using the "verkey" as the name of the Askar record. Operations like signing or encrypting DIDComm messages retrieve the key by verkey. Usually, when initiating a cryptographic operation, the key is looked up by first retrieving the `DIDInfo` object by DID (or by nym) from the wallet and then the `DIDInfo.verkey` value is used to retrieve the key.
+
+### Limitations
+
+- **Indy-Centric Design:** The current structure is closely tied to Indy, making it less suitable for other DID Methods.
+- **Single Key per DID:** Assumes a one-to-one relationship between DIDs and keys, which is not the case for many DID Methods.
+- **Inefficient for Multiple Keys:** Lacks support for DIDs with multiple verification methods and keys.
+
+## Proposed Updates
+
+### Goals
+
+- **Support Multiple Keys per DID:** Allow DIDs to have multiple associated keys and verification methods.
+- **Method-Agnostic Design:** Create a storage structure that supports various DID Methods beyond Indy.
+- **Efficient Key Retrieval:** Enable efficient lookup of keys based on DIDs, verification method IDs, verification relationships, and key types.
+
+### Key Storage
+
+#### Unbound Keys
+
+Unbound keys are keys not (or not yet) associated with a specific DID.
+
+Some DID Methods require knowledge of key material prior to creation of the DID. For example, in did:peer, the key material contributes directly to the formation of the DID itself. Unbound keys enable us to create and reference the key material during this early phase of DID creation for methods such as did:peer.
+
+Additionally, there may be use cases that we have not yet identified that may be best served by creating and referencing keys not associated with a DID at all. Unbound keys serve this purpose as well.
+
+- **Category:** `key`
+- **Name:** Multikey representation (e.g., `z6Mkw...`)
+- **Value:** The key material (private or public key)
+- **Tags:**
+  - `KeyAlg`: Implicit tag indicating the key algorithm (e.g., `ed25519`)
+  - `alias`: A list of human-friendly aliases for the key
+
+#### Bound Keys
+
+Bound keys are associated with a specific DID.
+
+When an unbound key is used to generate a DID for a method like did:peer, bound key representations MUST be stored in the wallet. The unbound representation MAY be removed after the bound representation is added.
+
+- **Category:** `key`
+- **Name:** Verification Method ID (absolute DID URL, e.g., `did:example:123#key-1`)
+- **Value:** The key material
+- **Tags:**
+  - `KeyAlg`: Implicit tag indicating the key algorithm (e.g. `ed25519`)
+  - `did`: The DID associated with the key
```

Comment:
is this limited to 1 did or can it be a list of dids?

Reply:
The intent is for this one to be a single DID. If this key is bound to multiple DIDs, there will be multiple key records, each one uniquely identified by the verification method ID of the key in that DID's document.